devices/HumanStateProvider/HumanStateProvider.cpp

```diff
@@ -654,6 +670,221 @@ bool HumanStateProvider::open(yarp::os::Searchable& config)
 
     pImpl->baseTransformSolution.setRotation(iDynTree::Rotation::Identity());
 
+    // ================================================
+    // INITIALIZE CUSTOM CONSTRAINTS FOR INTEGRATION-IK
+    // ================================================
+    yarp::os::Bottle& constraintGroup = config.findGroup("CUSTOM_CONSTRAINTS");
+    if (constraintGroup.isNull()) {
+        yError() << LogPrefix << "Failed to find group CUSTOM_CONSTRAINTS";
+        return false;
+    }
+    pImpl->customConstraintMatrix.resize(0, 0);
+    pImpl->customConstraintVariables.resize(0);
+    pImpl->customConstraintLowerBound.resize(0);
+    pImpl->customConstraintUpperBound.resize(0);
+    pImpl->customConstraintVariablesIndex.resize(0);
+    pImpl->custom_jointsVelocityLimitsNames.resize(0);
+    pImpl->custom_jointsVelocityLimitsValues.resize(0);
+    pImpl->custom_jointsVelocityLimitsIndexes.resize(0);
+    pImpl->k_u = 0.0;
+    pImpl->k_l = 0.0;
+
+    yInfo() << "==================>>>>>>> constraint group: " << constraintGroup.size();
+
+    for (size_t i = 1; i < constraintGroup.size(); i++) {
+        yInfo() << "group " << i;
+        if (!(constraintGroup.get(i).isList() && constraintGroup.get(i).asList()->size() == 2)) {
+            yError() << LogPrefix << "Childs of CUSTOM_CONSTRAINTS must be lists of two elements";
+            return false;
+        }
+        else {
+            yInfo() << "Everything is fine...";
+        }
+        yarp::os::Bottle* constraintList = constraintGroup.get(i).asList();
+        std::string constraintKey = constraintList->get(0).asString();
+        yarp::os::Bottle* constraintListContent = constraintList->get(1).asList();
+        yInfo() << constraintKey;
+        if (constraintKey == "custom_joints_velocity_limits_names") {
+
+            for (size_t i = 0; i < constraintListContent->size(); i++) {
+                pImpl->custom_jointsVelocityLimitsNames.push_back(
+                    constraintListContent->get(i).asString());
+            }
+            yInfo() << "custom_joints_velocity_limits_names: ";
+            for (size_t i = 0; i < pImpl->custom_jointsVelocityLimitsNames.size(); i++) {
+                yInfo() << pImpl->custom_jointsVelocityLimitsNames[i];
+            }
+        } // another option
+        else if (constraintKey == "custom_joints_velocity_limits_values") {
```

Comment:
I think to those joint velocity limits could replace the joint velocity limits that are imposed with [`integrationBasedJointVelocityLimit `](https://github.com/robotology/human-dynamics-estimation/blob/devel/devices/HumanStateProvider/HumanStateProvider.cpp#L521) option. So we should remove this deprecated option.

Reply:
The goal of this option is slightly different. Some joints of a robot may have different velocity limit for some specific joint. So with this option, we will overwrite the `general` joint velocity limit ( defined already by `integrationBasedJointVelocityLimit`) for a specific joint names defined by `custom_joints_velocity_limits_names` with the value `custom_joints_velocity_limits_values`.

Reply:
> So with this option, we will overwrite the `general` joint velocity limit

Ok, we should somehow document this then